### PR TITLE
Cleanup namespaces/restore drive after all tests are done

### PR DESCRIPTION
**Pull request by psjoshi789**
This adds the change to clean up all namespaces using the `factory_reset()` utility method.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -88,6 +88,11 @@ def write_report(tests, drive, output_path):
 
     r.close()
 
+def restore_drive(config):
+    drive = config['drive']['name']
+    tree = n_utils.generate_resource_tree()
+    n_utils.factory_reset(tree[drive]['sn'].strip())
+    return
```
Comment (atthorst) [inline src/main.py]: Redundant.  Can just remove.

File: src/main.py [hunk 2]
```diff
@@ -88,6 +88,11 @@ def write_report(tests, drive, output_path):
 
     r.close()
 
+def restore_drive(config):
+    drive = config['drive']['name']
+    tree = n_utils.generate_resource_tree()
```
Comment (atthorst) [inline src/main.py:100]: Do you need a call to sedutil to do a reset?

File: src/main.py [hunk 3]
```diff
@@ -121,10 +126,13 @@ def main():
         else:
             logger.info(f"Ignoring test: {test.name()}")
 
+    # cleanup any namespaces on the drive
```
Comment (atthorst) [inline src/main.py]: Do the previous tests need to be wrapped in a try/except block, and then the restore drive be in the except block as well?  

We kind of need to make sure it's run no matter if the tests passed or failed.  And I think tests should be assumed to be able to throw a failure.

Comment (psjoshi789) [reply src/main.py]: Agreed, the drive should be restored irrespective of whether the tests passed or failed.

File: src/main.py (hunk 2 above)
Comment (psjoshi789) [reply src/main.py:100]: Thanks, I will add a reset.

File: src/main.py [hunk 4]
```diff
@@ -88,6 +89,16 @@ def write_report(tests, drive, output_path):
 
     r.close()
 
+def restore_drive(config):
+    drive = config['drive']['name']
+    psid = config['drive']['psid']
+    logger.info(f"Cleaning up drive {drive} through a format.")
+
+    # Start with a PSID reset, just in case it's in a weird state
+    sedutil.reset_via_psid(drive, psid)
```
Comment (atthorst) [inline src/main.py]: Can we change this to only do this _if_ the PSID is set.  There could be a tolerance file where a PSID is not actually set.

Comment (atthorst) [reply src/main.py]: In case you're wondering - think SED but not OPAL/RUBY

Comment (psjoshi789) [reply src/main.py]: makes sense, I updated the code. Thanks.

**Review by atthorst — APPROVED** [review]

**Pull request merged**